src/PlaywrightSharp.Chromium/ChromiumNetworkManager.cs

```diff
@@ -23,5 +40,188 @@ internal Task SetUserAgentAsync(string userAgent)
         internal void InstrumentNetworkEvents(ChromiumSession session)
         {
         }
+
+        internal Task SetCacheEnabledAsync(bool enabled)
+        {
+            _userCacheDisabled = !enabled;
+            return UpdateProtocolCacheDisabledAsync();
+        }
+
+        private async void Client_MessageReceived(object sender, IChromiumEvent e)
+        {
+            try
+            {
+                switch (e)
+                {
+                    case NetworkResponseReceivedChromiumEvent networkResponseReceived:
+                        OnResponseReceived(networkResponseReceived);
+                        break;
+                    case NetworkRequestWillBeSentChromiumEvent networkRequestWillBeSent:
+                        OnRequestWillBeSent(networkRequestWillBeSent);
+                        break;
+                    case FetchRequestPausedChromiumEvent fetchRequestPaused:
+                        await OnRequestPausedAsync(fetchRequestPaused).ConfigureAwait(false);
+                        break;
+                    case NetworkLoadingFinishedChromiumEvent networkLoadingFinished:
+                        OnLoadingFinished(networkLoadingFinished);
+                        break;
+                }
+            }
+            catch (Exception ex)
+            {
+                // TODO Add Logger
+                /*
+                var message = $"Page failed to process {e.MessageID}. {ex.Message}. {ex.StackTrace}";
+                _logger.LogError(ex, message);
+                */
+                System.Diagnostics.Debug.WriteLine(ex.Message);
+                _client.OnClosed(ex.Message);
+            }
+        }
+
+        private void OnLoadingFinished(NetworkLoadingFinishedChromiumEvent e)
+        {
+            // For certain requestIds we never receive requestWillBeSent event.
+            // @see https://crbug.com/750469
+            if (!_requestIdToRequest.TryGetValue(e.RequestId, out var request))

```

Comment:
consider using `TryRemove` and delete the `_requestIdToRequest.Remove(request.RequestId);` call